Citus fixes

tanderegg opened the pull request.

tanderegg committed:
Add proper setup of PGDATA.

tanderegg committed:
Remove privilege escalation for postgres user.

tanderegg committed:
Ensure postgres is started once Citus install completes.

tanderegg committed:
Add additional package to install for postgres.

tanderegg committed:
Make the password setting tasks more robust.

tanderegg committed:
Add pause before attempting to connect.

tanderegg committed:
Added handler to role.

tanderegg committed:
Corrects .pgpass file placement.

tanderegg commented:
@rioh Can you review?  Thanks!

rioh commented on deploy/roles/postgresql-server-config/tasks/main.yml:
```diff
@@ -16,13 +16,16 @@
+    regexp: '^host(.*)postgres(.*)md5'
     backrefs: yes
-    line: '\1::1/128\2trust'
+    line: 'host\1postgres\2trust'
```
Will these work locally?

tanderegg replied:
Added a line to the defaults for pg_hba.conf to make this work in Vagrant.

rioh commented on deploy/roles/shared/handlers/main.yml:
```diff
@@ -31,13 +31,17 @@
+#  become: yes
+#  service:
+#    name: "{{ pp_servicename }}"
+#    state: restarted
```
Would this work if we were to deploy postgres without postgres-config (--tags=postgres) or that wouldn't happen?

tanderegg replied:
Handlers should run even though they don't have the tag, as long as a task that does have the tag is marked "changed"

tanderegg committed:
Add explicity pg_hba.conf setting for the 'postgres' user

tanderegg committed:
Added handler to postgres_server role

tanderegg committed:
Remote template1 from citus list

Installing the Citus extension on template1 creates a persistent connection from the Citus background worker, thus preventing any databases from being created.
